pkg/remoteconfig/updater.go

```diff
@@ -63,6 +65,7 @@ type DatadogAgentRemoteConfig struct {
 	ID            string                       `json:"id,omitempty"`
 	Name          string                       `json:"name,omitempty"`
 	CoreAgent     *CoreAgentFeaturesConfig     `json:"config,omitempty"`
+	ClusterAgent  *ClusterAgentFeaturesConfig  `json:"cluster_agent,omitempty"`
```

Comment:
This structure is unmarshalled from the RawConfig here and 
https://github.com/DataDog/datadog-operator/blob/03611c4e0472e9f993aa07d30b9fe051207a2fc9/pkg/remoteconfig/updater.go#L330-L331

And this config won't contain ClusterAgentFeatureConfig so curious why is this added here?

Reply:
Sorry if my question wasn't clear. I'm confused why the `ClusterAgentFeaturesConfig` is added here.

This structure
https://github.com/DataDog/datadog-operator/blob/03611c4e0472e9f993aa07d30b9fe051207a2fc9/pkg/remoteconfig/updater.go#L64-L71

 is for RC product `ProductAgentConfig` supporting CoreAgent, SystemProbe and SecurityAgent features. It's subscribed here

https://github.com/DataDog/datadog-operator/blob/03611c4e0472e9f993aa07d30b9fe051207a2fc9/pkg/remoteconfig/updater.go#L181

While 

https://github.com/DataDog/datadog-operator/blob/03611c4e0472e9f993aa07d30b9fe051207a2fc9/pkg/remoteconfig/updater.go#L183

subscribes a different `ProductOrchestratorK8sCRDs`. So I'm curious why it's put under `DatadogAgentRemoteConfig` structure instead of having it's own structure?